Create_Group.py

```diff
@@ -46,48 +53,80 @@ def __init__(self, parent=None):
         completer = QCompleter()
         edit.setCompleter(completer)
         model = QStringListModel()
         completer.setModel(model)
         self.get_data(model,Tag_list)
 
+        line_g = self.lineEdit_GroupName
+        regex = QtCore.QRegExp("[a-z-A-Z-0-9-@-#-$_]+")
+        validator = QtGui.QRegExpValidator(regex)
+        line_g.setValidator(validator)
+
+        line_update = self.lineEdit_UpdateRate
+        regex2 = QtCore.QRegExp("[0-9]+")
+        validator2 = QtGui.QRegExpValidator(regex2)
+        line_update.setValidator(validator2)
+
         self.comboBox.activated[str].connect(self.onActivate)
         self.button_cancel.clicked.connect(QtCore.QCoreApplication.instance().quit)
         self.button_createGroup.clicked.connect(self.load_tree)
         self.button_searchTag.clicked.connect(self.search)
         self.button_deleteTag.clicked.connect(self.delete_Tag)
         self.connect(self.treeWidget, QtCore.SIGNAL("itemDoubleClicked(QTreeWidgetItem *,int)"), self.check_condition)
 
-    #Tag_list = opc.list('*',recursive=True)
     def get_data(self,model,Tag_list):
-        print Tag_list
-        model.setStringList = (Tag_list)
+        model.setStringList(Tag_list)
 
-    def search(self):
+    def search(self,Tag_list):
         input_tag = self.lineEdit_searchTag.text()
-        if any(input_tag in sublist for sublist in Sub_Tags):
+        check_list = opc.list('*',recursive=True)
+        if input_tag in check_list:
             result = QMessageBox.question(self, 'Message', "The tag has been Found !! \n Do you want to insert it ?", QMessageBox.Yes | QMessageBox.No, QMessageBox.No)
             if result == QMessageBox.Yes:
-                self.Add_to_tree(input_tag)
-            else:
-                print "No"
+                self.Add_to_tree(str(input_tag))
         else:
             QMessageBox.about(self, "Sucess", "Enter a valid Tag")
 
     def load_tree(self):
         group = self.lineEdit_GroupName.text()
         rate = self.lineEdit_UpdateRate.text()
         group = str(group)
-        if not group:
-            QMessageBox.about(self, "Error", "Enter the Group name!!")
+        if not group or not Tarray :
+            QMessageBox.about(self, "Error", "Check the Group name / Tags !!")
         else:
-             if not rate:
-                QMessageBox.about(self, "Error", " Update rate is 5 seconds!!")
-                rate = "5"
-             self.create_group(group, rate)
+            if group not in group_list:
+                if not rate:
+                    QMessageBox.about(self, "Error", " Update rate is 5 seconds!!")
+                    rate = "5"
+                else:
+                    try:
+                        val = int(rate)
+                        if type(val) == int:
+                            self.create_group(group, rate)
+                    except ValueError:
+                        QMessageBox.about(self, "Error", "Enter the a Integer for Update Rate!!")

```

Comment:
?

Reply:
checking whether the data input in the text box is integer or not